crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -189,11 +219,166 @@ impl Consensus {
     }
 
     #[instrument(skip(self))]
-    async fn commit(&mut self) -> Result<response::Commit, BoxError> {
+    async fn commit(&mut self) -> anyhow::Result<response::Commit> {
         let app_hash = self.app.commit(self.storage.clone()).await;
         Ok(response::Commit {
             data: app_hash.0.to_vec().into(),
             ..Default::default()
         })
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::str::FromStr;
+
+    use ed25519_consensus::SigningKey;
+    use prost::Message as _;
+    use rand::rngs::OsRng;
+    use tendermint::{
+        account::Id,
+        Hash,
+        Time,
+    };
+
+    use super::*;
+    use crate::{
+        accounts::types::Nonce,
+        sequence,
+        transaction::{
+            action::Action,
+            Unsigned,
+        },
+    };
+
+    fn new_prepare_proposal_request(txs: Vec<Bytes>) -> request::PrepareProposal {
+        request::PrepareProposal {
+            txs,
+            max_tx_bytes: 1024,
+            local_last_commit: None,
+            misbehavior: vec![],
+            height: 1u32.into(),
+            time: Time::now(),
+            next_validators_hash: Hash::default(),
+            proposer_address: Id::from_str("0CDA3F47EF3C4906693B170EF650EB968C5F4B2C").unwrap(),
+        }
+    }
+
+    fn new_process_proposal_request(txs: Vec<Bytes>) -> request::ProcessProposal {
+        request::ProcessProposal {
+            txs,
+            proposed_last_commit: None,
+            misbehavior: vec![],
+            hash: Hash::default(),
+            height: 1u32.into(),
+            next_validators_hash: Hash::default(),
+            time: Time::now(),
+            proposer_address: Id::from_str("0CDA3F47EF3C4906693B170EF650EB968C5F4B2C").unwrap(),
+        }
+    }
+
+    #[test]
+    fn prepare_proposal() {
```

Comment:
Is this test useful? All it's doing is running `generate_transaction_commitment` in- and outside of `Consensus::prepare_proposal`, so it's only asserting that it's indeed following the individual steps.

Alternative suggestion: run a proposal through `prepare_proposal` and then through `process_proposal`, because these must always match.

Reply:
I think we want to do a snapshot test here. That is we define the input and compare against an expected output which is saved.

If we change commitment generating it will break and either be on purpose or not, but it's always a breaking change. We need to know if the commitment generating logic changes though so it's a useful test.

We can wrap this manually or use a framework and have proper snapshots [insta for rust](https://docs.rs/insta/latest/insta/)). 

I think the roundtrip is also something we should test, because they do need to match.

Reply:
I updated this test to also send the resulting proposal through `handle_process_proposal` as well and make sure it passes